stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/client/WebSocketClientHandshakeListener.java

```diff
@@ -26,53 +26,55 @@
-    private final boolean readyOnConnect;
     private final ObjectValue webSocketClient;
-    private CountDownLatch countDownLatch;
+    private static final Logger logger = LoggerFactory.getLogger(WebSocketClientHandshakeListener.class);
 
-    public WebSocketClientHandshakeListener(ObjectValue webSocketClient,
-                                            WebSocketService wsService,
-                                            WebSocketClientConnectorListener clientConnectorListener,
-                                            boolean readyOnConnect, CountDownLatch countDownLatch) {
+    public WebSocketClientHandshakeListener(ObjectValue webSocketClient, WebSocketService wsService,
+                                            WebSocketClientListener clientConnectorListener, boolean readyOnConnect) {
+        super(webSocketClient, wsService, clientConnectorListener, readyOnConnect);
         this.webSocketClient = webSocketClient;
-        this.wsService = wsService;
-        this.clientConnectorListener = clientConnectorListener;
-        this.readyOnConnect = readyOnConnect;
-        this.countDownLatch = countDownLatch;
     }
 
     @Override
     public void onSuccess(WebSocketConnection webSocketConnection, HttpCarbonResponse carbonResponse) {
-        //using only one service endpoint in the client as there can be only one connection.
-        webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD,
-                            HttpUtil.createResponseStruct(carbonResponse));
-        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_HTTP_PKG_ID,
-                                                                           WebSocketConstants.WEBSOCKET_CONNECTOR);
-        WebSocketOpenConnectionInfo connectionInfo = getWebSocketOpenConnectionInfo(webSocketConnection,
-                                                                                    webSocketConnector);
-        WebSocketUtil.populateWebSocketCaller(webSocketConnection, webSocketClient);
-        clientConnectorListener.setConnectionInfo(connectionInfo);
-        if (readyOnConnect) {
-            WebSocketUtil.readFirstFrame(webSocketConnection, webSocketClient);
+        super.onSuccess(webSocketConnection, carbonResponse);
+        logger.debug(CONNECTED_TO + webSocketClient.getStringValue(WebSocketConstants.
+                CLIENT_URL_CONFIG));
+        // Following these are created for future connection
+        // Check whether the config has retry config or not
+        // It has retry config, set these variables to default variable
+        RetryContext retryConfig;
+        if (hasRetryConfig(webSocketClient)) {
+            retryConfig = (RetryContext) webSocketClient.getNativeData(RETRY_CONFIG);
+            setWebSocketEndpoint(retryConfig, webSocketClient, webSocketConnection);
+            // Read the next frame when readyOnConnect is true in first the connection or after the first connection
```

Comment:
Is this the first connection or is it the next connection? We have a call to read frame in the super class as well. Can we write a better comment?